app/scripts/lib/modal_manager.js

```diff
@@ -0,0 +1,127 @@
+  'use strict';
+
+  function ModalManager () {
+    this._viewStates = [];
+    this.$modal = $('#modal');
+
+    // Listen for all clicks
+    $(document).on('click', _.bind(this._closeOnClick, this));
+
+    // Close modal on resize to prevent weirdness between desktop and mobile
+    $(document).on('resize', _.debounce(_.bind(this.close, this), 100));
+
+    // Handle default button actions
+    this.$modal.on('click', 'a.close', _.bind(this.close, this));
+    this.$modal.on('click', 'a.back', _.bind(this.pop, this));
+  }
+
+  _.extend(ModalManager.prototype, {
+    open: function (view, options) {
+      this._destroyViews();
+
+      this.push(view, options);
+    },
+
+    push: function (view, options) {
+      this._viewStates.push({ view: view, options: (options || {}) });
+
+      this._show();
+    },
+
+    pop: function () {
+      var viewState = this._viewStates.pop();
+
+      if (viewState) {
+        viewState.view.destroy();
+      }
+
+      if (this._viewStates.length > 0) {
+        this._show();
+      } else {
+        this._hide();
+      }
+    },
+
+    close: function (options) {
+      if (!options) {
+        options = {};
+      }
+
+      this._destroyViews();
+      this._hide(options);
+    },
+
+    _destroyViews: function () {
+      _.each(this.viewStates, function (viewState) {
+        viewState.view.destroy();
+      });
+
+      this._viewStates = [];
+    },
+
+    _show: function () {
+      var viewState = _.last(this._viewStates);
+
+      viewState.view.render();
+
+      // Force delegate events to fix an issue where restoring a previous view breaks event bindings
+      viewState.view.delegateEvents();
+
+      // TODO: Add fancy positioning: likely in the middle of the screen. We're only supporting
+      // full screen right now, so it can come later.
+
+      // Add full-screen class if the full screen option is enabled
+      if (viewState.options.fullScreen) {
+        this.$modal.addClass('full-screen');
+      } else {
+        this.$modal.removeClass('full-screen');
+      }
+
+      // Replace modal contents
+      this.$modal.html(viewState.view.el);
+
+      // Animate if an effect is provided
+      if (viewState.options.effect) {
+        this.$modal.velocity(viewState.options.effect);
+      } else {
+        this.$modal.show();
+      }
+    },
+
+    _hide: function (options) {
+      // Animate if an effect is provided
+      if (options.effect) {
+        this.$modal.velocity(options.effect, function () {
+          // Reset styles and hide
+          $(this).removeAttr('style').hide();
+        });
+      } else {
+        this.$modal.hide();
+      }
+    },
+
+    _closeOnClick: function (event) {
+      if (this.$modal.is(':visible')) {
+        var $target = $(event.target);
+
+        // Close if target isn't modal, modal isn't the parent, and still in the dom
+        if (!$target.is('#modal') && !$target.parents('#modal').length && $target.parents('body').length) {
```

Comment:
(Sorry, I have a lot of questions - been a little while since I did front-end stuff.)

Hmm. So, "if this.modal is visible, but the target is not modal or one of its descendants, and the click is on a descendant of body, then close". I don't see how the click target could not be modal, since it's full-screen. I guess this is for the case where the modal is *not* full-screen? I guess let's just leave this code in here, although it's awkward that we don't have an underlay to provide a lightbox or whitebox effect underneath this.modal. Maybe add a todo about inserting that underlay div when we want a non-full-screen modal?